In [16]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from scipy import stats

In [4]:
students= pd.read_csv('students.csv')

Our job is to run a hypothesis testing over weight column of the data set. The average weight of european adult is 70.8 kg. The standard deviation is given 8.63. This can be done using Z-Test

In [6]:
weights= students.weight

Take a random sample of size 30 and calculate the sample mean.

In [14]:
sample= np.random.choice(weights, 30)
x_bar= np.mean(sample)
x_bar

73.14

$\LARGE{Hypothesis~Testing}$ <br>
$\Large{Critical~Value ~Approach}$

$H_{0}$: The average weight of the students is equal to the average weight of European adults ie $\mu= 70.8$<br>
$H_1:\mu\neq70.8$ <br>
Significance level $\alpha= 0.05$<br>
Test statistic $z= \frac{\bar{x}-\mu}{\sigma/\sqrt{n}}$

In [15]:
#Calculate test statistic
z= (x_bar- 70.8)/ (8.63/np.sqrt(30))
z

1.4851341651936159

The test will be a two tailed test.<br>
$H_{A_1}:\mu\neq 70.8$<br>
$H_{A_2}:\mu< 70.8$<br>
$H_{A_3}:\mu> 70.8$<br>
Calculate the critical values accordingly for all the different hypothesis.

In [21]:
alpha= 0.05

In [23]:
z_Ha1= stats.norm.ppf(1- alpha/2)
z_Ha2= stats.norm.ppf(alpha)
z_Ha3= stats.norm.ppf(1- alpha)
z_Ha1, z_Ha2, z_Ha3

(1.959963984540054, -1.6448536269514729, 1.6448536269514722)

Critical value for Ha1= $\pm 1.959963984540054$ <br>
Ha2= -1.6448536269514729<br>
ha3= 1.6448536269514729 <br>
For hypothesis 1: -1.95<1.48< 1.95<br>
For hypothesis 2: -1.64<1.48 <br>
For hypothesis 3: 1.48< 1.64 <br>
Therefore, from all the above results we can conlude that the two tail test accepts the null hypothesis with significance level 0.05. 

$\Large{P\ Value\ Approach}$

In [43]:
# Perform the above steps upto hypothesis fromation. Calculate p values for all test statistic of the hypothesis.
p_Ha1= (1- stats.norm.cdf(z/2))*2 # Two tail test
p_Ha2= stats.norm.cdf(-z) # Left tail test
p_Ha3= 1- stats.norm.cdf(z) # Right tail test

In [44]:
p_Ha1, p_Ha2, p_Ha3

(0.4577438238894378, 0.06875414366085851, 0.06875414366085852)

All the p values are greater than $\alpha$ we cannot reject null hypothesis.